dsgrid/config/time_dimension_base_config.py

```diff
@@ -110,3 +123,111 @@ def list_expected_dataset_timestamps(self):
             List of tuples of columns representing time in the load_data table.
 
         """
+
+    def _convert_time_to_project_time_interval(
+        self, df=None, project_time_dim=None, wrap_time=True
+    ):
+        """Shift time to match project time based on TimeIntervalType
+        If time range spills over into another year after time interval alignment,
+        the time range will be wrapped around so it's bounded within the year
+        """
+        if project_time_dim is None:
+            return df
+
+        dtime_interval = self.get_time_interval_type()
+        ptime_interval = project_time_dim.get_time_interval_type()
+        time_col = project_time_dim.get_timestamp_load_data_columns()
+
+        assert len(time_col) == 1, time_col
+        time_col = time_col[0]
+
+        if dtime_interval != ptime_interval:
+            match (dtime_interval, ptime_interval):
+                case (TimeIntervalType.PERIOD_BEGINNING, TimeIntervalType.PERIOD_ENDING):
+                    df = df.withColumn(
+                        time_col,
+                        F.col(time_col)
+                        + F.expr(f"INTERVAL {self.get_frequency().seconds} SECONDS"),
+                    )
+                case (TimeIntervalType.PERIOD_ENDING, TimeIntervalType.PERIOD_BEGINNING):
+                    df = df.withColumn(
+                        time_col,
+                        F.col(time_col)
+                        - F.expr(f"INTERVAL {self.get_frequency().seconds} SECONDS"),
+                    )
+
+        if wrap_time:
+            df = self._apply_time_wrap(df, project_time_dim)
+
+        return df
+
+    @staticmethod
+    def _apply_time_wrap(df, project_time_dim):
+        """If dataset_time does not match project_time, apply time-wrapping"""
+
+        time_col = project_time_dim.get_timestamp_load_data_columns()
+        assert len(time_col) == 1, time_col
+        time_col = time_col[0]
+
+        project_time = set(
```

Comment:
This creates a list and then another list and then a set. You could drop the middle list.
```
{row[0] for row in project_time_dim ...}
```